File: examples/python/simulations.py
"""Installment and value simulations — see rules/python/simulations.md."""

from twila_parcelemais import (
    CalculationValueType,
    ParceleMaisClient,
    SimulateInstallmentsRequest,
    SimulateValuesRequest,
)


def simulate_installments(client: ParceleMaisClient, amount: float) -> None:
    options = client.simulations.simulate_installments(
        SimulateInstallmentsRequest(requested_amount=amount, calculation_value_type=CalculationValueType.GROSS_AMOUNT)
    )
    for option in options:
        print(f"{option.term}x de R$ {option.installment_amount:.2f} (total R$ {option.total_amount:.2f})")


def simulate_values(client: ParceleMaisClient, amount: float, term: int) -> None:
    values = client.simulations.simulate_values(SimulateValuesRequest(amount=amount, term=term))
    print(f"Valor de venda: R$ {values.sale_amount:.2f}")
    print(f"Valor de desembolso: R$ {values.disbursement_amount:.2f}")
    print(f"Valor da parcela: R$ {values.installment_amount:.2f}")

File: examples/php/simulations.php
<?php

declare(strict_types=1);

use Twila\ParceleMais\Config\ClientOptions;
use Twila\ParceleMais\Config\Environment;
use Twila\ParceleMais\ParceleMaisClient;
use Twila\ParceleMais\Simulations\CalculationValueType;
use Twila\ParceleMais\Simulations\SimulateInstallmentsRequest;
use Twila\ParceleMais\Simulations\SimulateValuesRequest;

$client = new ParceleMaisClient(new ClientOptions(
    getenv('PARCELEMAIS_CLIENT_ID'),
    getenv('PARCELEMAIS_CLIENT_SECRET'),
    Environment::STAGING
));

// Simulate installments for a requested amount
$installments = $client->simulations->simulateInstallments(
    new SimulateInstallmentsRequest(1500.00, CalculationValueType::GROSS_AMOUNT)
);
foreach ($installments as $installment) {
    printf(
        "%dx de R$ %.2f (total R$ %.2f)\n",
        $installment->term,
        $installment->installmentAmount,
        $installment->totalAmount
    );
}

// Simulate values for a specific term
$values = $client->simulations->simulateValues(new SimulateValuesRequest(1500.00, 12));
printf(
    "Valor de venda: R$ %.2f | Desembolso: R$ %.2f | Parcela: R$ %.2f\n",
    $values->saleAmount,
    $values->disbursementAmount,
    $values->installmentAmount
);
